### PR TITLE
Implement structured logging

### LINKED ISSUE #1285: Please Make Event Logging Verbosity Configurable
**Which component**:
controller

**Is your feature request related to a problem? Please describe.**
Unless it's tucked away someplace not obvious in documentation, it doesn't look like there is an `arg` that can be set on the Deployment template spec nor in the helm chart. This would be greatly appreciated as the controller can be extremely chatty when it iterates on decrypting SealedSecret resources, particularly in large Kubernetes clusters.

Such normal events would be considered INFO, which are usually not very interesting for day-to-day knowledge when you may instead be interested specifically in WARN or more severe.

Here's an example event log from the controller:
```
2023/07/27 22:11:21 Updating staging/account-secrets
2023/07/27 22:11:21 Event(v1.ObjectReference{Kind:"SealedSecret", Namespace:"staging", Name:"account-secrets", UID:"45ef60cc-c7bf-4508-a861-a0514dce9c1c", APIVersion:"bitnami.com/v1alpha1", ResourceVersion:"768125925", FieldPath:""}): type: 'Normal' reason: 'Unsealed' SealedSecret unsealed successfully 
```

And worse, some events that aren't timestamped. https://github.com/bitnami-labs/sealed-secrets/discussions/1274 is a good example of this.

**Describe the solution you'd like**
A flag that can be set in `args` on the Deployment template spec and helm chart. Also if implemented please document clearly what the valid values are. Some tools abide by the native Kubernetes logging verbosities which are integers (0 = shut up, 1 = info, etc.) while others support logging verbosity but only note the flag expects a string type.

**Describe alternatives you've considered**
Are there any?

**Additional context**
This might take more effort to implement because none of the events produced by the controller are classified classically - e.g.: `INFO`, `DEBUG`, `WARN`, etc..

### PR DESCRIPTION
**Pull request by mohamed-essam**
**Description of the change**

Change controller logging to use `log/slog`.

**Benefits**

1. Allow structured logging.
2. Allow changing logging format between text and JSON.
3. Allow changing logging level.

**Possible drawbacks**

N/A

**Applicable issues**

- fixes #1285 

**Additional information**

* I took the liberty of removing log at line 91 in pkg/controller/main.go as it only served as a separator
* To implement the functionality of `--log-info-to-stdout`, I had to implement a custom handler that would hand logs down to one of two slog.Loggers in place of the package in `pkg/log` that handled this previously

### CODE REVIEW

File: helm/sealed-secrets/templates/deployment.yaml [hunk 1]
```diff
@@ -120,6 +120,14 @@ spec:
             {{- if .Values.logInfoStdout }}
             - --log-info-stdout
             {{- end }}
+            {{- if .Values.logLevel }}
+            - --log-level
+            - {{ .Values.logLevel }}
+            {{- end }}
+            {{- if .Values.logFormat }}
+            - --log-level
```
Comment (tewfik-ghariani) [inline helm/sealed-secrets/templates/deployment.yaml:128]: log-format* ^^

Comment (mohamed-essam) [reply helm/sealed-secrets/templates/deployment.yaml]: Thanks for spotting this :pray:

**Review by alemorcuq — CHANGES_REQUESTED** [review]
Thanks @mohamed-essam! Please, check my comments. The PR looks awesome.

File: pkg/controller/main.go [hunk 2]
```diff
@@ -142,8 +144,17 @@ func initKeyRenewal(ctx context.Context, registry *KeyRegistry, period, validFor
 
 func Main(f *Flags, version string) error {
 	registerMetrics(version)
+
+	// Set logging
```
[the remark this answers is not in the document]
Comment (mohamed-essam) [reply pkg/controller/main.go]: Thanks @alemorcuq for your observations :pray:

File: pkg/controller/main.go [hunk 3]
```diff
@@ -142,8 +144,17 @@ func initKeyRenewal(ctx context.Context, registry *KeyRegistry, period, validFor
...
+	logLevel := slog.Level(0)
+	(&logLevel).UnmarshalText([]byte(f.LogLevel))
```
[the remark this answers is not in the document]
Comment (mohamed-essam) [reply pkg/controller/main.go]: For some reason it was behaving weirdly with my IDE for a while but apparently I had an issue with gopls

File: cmd/controller/main.go [hunk 4]
```diff
@@ -85,14 +88,26 @@ func mainE(w io.Writer, fs *flag.FlagSet, gofs *goflag.FlagSet, args []string) e
...
+	if flags.LogInfoToStdout {
+		slog.SetDefault(slog.New(log.New(os.Stdout, os.Stderr, flags.LogFormat, opts)))
+	} else {
+		slog.SetDefault(slog.New(log.New(os.Stderr, os.Stderr, flags.LogFormat, opts)))
+	}
+
 	ssv1alpha1.AcceptDeprecatedV1Data = flags.AcceptV1Data
 
-	fmt.Fprintf(w, "controller version: %s\n", VERSION)
+	slog.Info("controller version", "version", VERSION)
```
[the remark this answers is not in the document]
Comment (alemorcuq) [reply cmd/controller/main.go:105]: I agree, we can move it to the if condition. The version is already being printed in the logs when starting the container 👍

File: pkg/controller/server.go [hunk 5]
```diff
@@ -48,14 +49,14 @@ func httpserver(cp certProvider, sc secretChecker, sr secretRotator, burst int,
 	mux.Handle("/v1/verify", Instrument("/v1/verify", httpRateLimiter.RateLimit(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		content, err := io.ReadAll(r.Body)
 		if err != nil {
-			log.Errorf("Error handling /v1/verify request: %v", err)
+			slog.Error("Error handling /v1/verify request: %v", err)
```
[the remark this answers is not in the document]
Comment (alemorcuq) [reply pkg/controller/server.go]: Could you please check all the `slog.Error` calls in this file? There are still some of them without the `error` field.

[the remark this answers is not in the document]
Comment (mohamed-essam) [reply pkg/controller/server.go]: Sorry misunderstood your comment as a missed line not lines, fixed.

**Pull request merged**